synapseclient/models/activity.py

```diff
@@ -1 +1,342 @@
-# TODO
+import asyncio
+from dataclasses import dataclass, field
+from typing import List, Dict, Union, Tuple
+
+from synapseclient.activity import Activity as Synapse_Activity
+from opentelemetry import trace, context
+
+from typing import Optional, TYPE_CHECKING
+
+from synapseclient import Synapse
+from synapseclient.core.async_utils import otel_trace_method
+
+if TYPE_CHECKING:
+    from synapseclient.models import File, Table
+
+USED_URL_CONCRETE_TYPE = "org.sagebionetworks.repo.model.provenance.UsedURL"
+USED_ENTITY_CONCRETE_TYPE = "org.sagebionetworks.repo.model.provenance.UsedEntity"
+
+tracer = trace.get_tracer("synapseclient")
+
+
+@dataclass
+class UsedEntity:
+    """
+    Reference to a Synapse entity that was used or executed by an Activity.
+
+    Arguments:
+        target_id: The ID of the entity to which this reference refers.
+        target_version_number: The version number of the entity to which this reference refers.
+    """
+
+    target_id: Optional[str] = None
+    """The the id of the entity to which this reference refers."""
+
+    target_version_number: Optional[int] = None
+    """The version number of the entity to which this reference refers."""
+
+
+@dataclass
+class UsedURL:
+    """
+    URL that was used or executed by an Activity.
```

Comment:
Maybe specify what this URL should be pointing to

Reply:
Adding this from the synapse help docs: "such as a link to a GitHub commit or a link to a specific version of a software tool."

Reply:
https://help.synapse.org/docs/Provenance.1972470373.html#Provenance-OverviewofSynapseProvenance